pandas/tests/test_reshape.py

```diff
@@ -171,6 +172,26 @@ def test_basic(self):
         expected.index = list('ABC')
         assert_frame_equal(get_dummies(s_series_index, sparse=self.sparse), expected)
 
+    def test_basic_types(self):
+        # GH 10531
+        s_list = list('abc')
+        s_series = Series(s_list)
+        s_df = DataFrame({'a': [0, 1, 0, 1, 2],
+                          'b': ['A', 'A', 'B', 'C', 'C'],
+                          'c': [2, 3, 3, 3, 2]})
+
+        exp_type = DataFrame if not self.sparse else SparseDataFrame
+        self.assertEqual(type(get_dummies(s_list, sparse=self.sparse)), exp_type)
+        self.assertEqual(type(get_dummies(s_series, sparse=self.sparse)), exp_type)
+
+        r = get_dummies(s_df, sparse=self.sparse, columns=s_df.columns)
+        self.assertEqual(type(r), exp_type)
+
+        r = get_dummies(s_df, sparse=self.sparse, columns=['a'])
+        has_sparse_blocks = any(type(b) == pd.core.internals.SparseBlock
```

Comment:
@jreback This is what r looks like:
```
   b  c  a_0  a_1  a_2
0  A  2    1    0    0
1  A  3    0    1    0
2  B  3    1    0    0
3  C  3    0    1    0
4  C  2    0    0    1
```
Only `a_0`, `a_1`, `a_2` will be in SparseBlocks. `b` and `c` remain dense.